File: Candy/python/No_2.py
# 編號：CANDY-002
# 程式語言：Python
# 題目：請寫一小段程式，印出連續陣列裡缺少的字元

chars_1 = ["a", "b", "c", "d", "f", "g"]
chars_2 = ["O", "Q", "R", "S"]


def missing_char(chars):
    parse_chars = map(lambda element: ord(element), chars)
    for i in parse_chars:
        if chr(i) != chars[(i - ord(chars[0]))]:
            return chr(i - 1)


print(missing_char(chars_1))  # // 印出 e
print(missing_char(chars_2))  # // 印出 P

File: No_2.js
// 編號：CANDY-002
// 程式語言：JavaScript
// 題目：請寫一小段程式，印出連續陣列裡缺少的字元

const chars1 = ["a", "b", "c", "d", "f", "g"];
const chars2 = ["O", "Q", "R", "S"];

function missingChar(chars) {
  const array = chars.map((n) => n.charCodeAt());
  let ans;
  for (let i = 0; i < array.length - 1; i++) {
    if (array[i] + 1 !== array[i + 1]) {
      ans = array[i] + 1;
    }
  }
  return String.fromCharCode(ans);
}

console.log(missingChar(chars1)); // 印出 e
console.log(missingChar(chars2)); // 印出 P

// 提示：
// 可使用字串的 charCodeAt 方法...
